override/uncity.go

```diff
@@ -136,3 +137,33 @@ func mountIndexer(defaultPath string) indexer {
 		}
 	}
 }
+
+func portIndexer(y any, p tree.Path) (string, error) {
+	switch value := y.(type) {
+	case map[string]any:
+		target, ok := value["target"].(int)
+		if !ok {
+			return "", fmt.Errorf("service ports %s is missing a target port", p)
+		}
+		published, ok := value["published"].(string)
+		if !ok {
+			return "", fmt.Errorf("service ports %s is missing a published port", p)
+		}
+		host, ok := value["host_ip"].(string)
+		if !ok {
+			host = "0.0.0.0"
+		}
+		protocol, ok := value["protocol"].(string)
+		if !ok {
+			protocol = "tcp"
+		}
+		mode, ok := value["mode"].(string)
```

Comment:
`mode` is not part of the key, this is actually the only "configuration" attribute for a port declaration